api/taskcluster/webhook.go

```diff
@@ -37,6 +39,92 @@ var (
 // Non-fatal error when there is no result (e.g. nothing finishes yet).
 var errNoResults = errors.New("no result URLs found in task group")
 
+// TaskInfo is an abstraction of a Taskcluster task, containing the necessary
+// information for us to process the task in wpt.fyi.
+type TaskInfo struct {
+	Name   string
+	TaskID string
+	State  string
+}
+
+// TaskGroupInfo is an abstraction of a Taskcluster task group, containing the
+// necessary information for us to process the group in wpt.fyi.
+type TaskGroupInfo struct {
+	TaskGroupID string
+	Tasks       []TaskInfo
+}
+
+// EventInfo is an abstraction of a GitHub Status event, containing the
+// necessary information for us to process the event in wpt.fyi.
+type EventInfo struct {
+	Sha     string
+	RootURL string
+	TaskID  string
+	Master  bool
+	Sender  string
+	Group   *TaskGroupInfo
+}
+
+// API is an interface for Taskcluster related methods.
+type API interface {
+	GetTaskGroupInfo(string, string) (*TaskGroupInfo, error)
+}
+
+type apiImpl struct {}
+
+// GetEventInfo turns a GitHub Status event payload into an EventInfo struct.
+//
+// It has two return values:
+//   * An EventInfo struct; this will be non-nil if and only if the event
+//     should be processed.
+//   * An error; if non-nil, then this event has caused an error that should be
+//     reported to the user.
+func GetEventInfo(payload []byte, log shared.Logger, api API) (*EventInfo, error) {
```

Comment:
Might be up for debate, but I would rather return `EventInfo`.  A pointer is not needed (*event used for processTaskclusterBuild) and returning an empty struct is not inherently bad. Once an error is returned, you shouldn't use non-error return values for anything else.

Reply:
ties to [this comment] (https://github.com/web-platform-tests/wpt.fyi/pull/1970/files#r434037566)